src/main/kotlin/solve/utils/nodes/listcell/dragdrop/DragAndDropListCell.kt

```diff
@@ -65,24 +77,34 @@ abstract class DragAndDropListCell<T> : ListCell<T>() {
             if (item == null) {
                 return@setOnDragDropped
             }
-            if (isListViewCellSource(event.gestureSource)) {
-                val droppedCellItem =
-                    (event.gestureSource as? DragAndDropListCell<Any>)?.item as? T ?: return@setOnDragDropped
-                val droppedCellIndex = listView.items.indexOf(droppedCellItem)
 
-                listView.items[droppedCellIndex] = item
-                listView.items[index] = droppedCellItem
-
-                val thisItemInfo = createItemInfo(item)
-                val droppedItemInfo = createItemInfo(droppedCellItem)
-                setOnDragDropped(event, thisItemInfo, droppedItemInfo)
-            }
+            onDragDropped(event)
 
             event.isDropCompleted = true
             event.consume()
         }
     }
 
+    fun onDragDropped(event: DragEvent) {
+        if (isListViewCellSource(listView, event.gestureSource)) {
+            val droppedCellItem =
+                (event.gestureSource as? DragAndDropListCell<Any>)?.item as? T ?: return
+            if (!isAbleToDropItem(item, droppedCellItem)) {
```

Comment:
Can we show this somehow on the UI?

Reply:
I've added a popover hint:
![layers_hint](https://user-images.githubusercontent.com/66365072/230470161-71e979a6-2ad7-4dd5-bd60-cbf8bfb4d04e.gif)